### PR TITLE
Added weapons to gta_data_service.

### PR DESCRIPTION
**Pull request by aa15032261**

### CODE REVIEW

**Review by Yimura — DISMISSED** [review]
Looks good, haven't tested yet.

**Review by karifeld — COMMENTED** [review]
works good, one thing. it looks like youve added the `WEAPON_TACTICALRIFLE` (service carbine) to the `hash_blacklist` array by mistake.

that gun is unlockable in-game by normal means, i dont know if this is intentional or not

**Comment by aa15032261** [comment]
> works good, one thing. it looks like youve added the `WEAPON_TACTICALRIFLE` (service carbine) to the `hash_blacklist` array by mistake.
> 
> that gun is unlockable in-game by normal means, i dont know if this is intentional or not

It was a mistake. Thanks for spotting that! It is fixed in the latest commit.

**Comment by Yimura** [comment]
@karifeld thanks for testing it.
@aa15032261 I'll merge it then after I've had one last look.

File: BigBaseV2/src/services/gta_data/gta_data_service.cpp [hunk 1]
```diff
@@ -85,27 +93,61 @@ namespace big
...
 
-	void gta_data_service::load_from_file(file file_to_load, file file_etag, std::string url, void(gta_data_service::* load_func)(), std::string data_name)
+		if (idx == -1)
+		{
+			return empty_weapon_item;
+		}
+		else
+		{
+			return m_weapon_item_arr[idx];
+		}
```
Comment (Yimura) [inline BigBaseV2/src/services/gta_data/gta_data_service.cpp]: redundant else as you've already returned (exit) out of the function in the if statement.

Instead you can just return at the end.

File: BigBaseV2/src/services/gta_data/weapon_item.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		}
+
+		if (item_json["Category"].is_null())
+		{
+			this->weapon_type = "NULL";
+		}
+		else
+		{
+			this->weapon_type = item_json["Category"];
+		}
```
Comment (Yimura) [inline BigBaseV2/src/services/gta_data/weapon_item.cpp]: You can probably default the this->weapon_type = "NULL" and if the Category isn't null assign it.
```cpp
this->weapon_type = "NULL";
if (!item_json["Category"].is_null())
{
	this->weapon_type = item_json["Category"];
}
```

**Review by Yimura — CHANGES_REQUESTED** [review]

**Review by Yimura — APPROVED** [review]

**Pull request merged**